src/dos/cdrom_ioctl_linux.cpp

```diff
@@ -83,26 +269,47 @@ bool CDROM_Interface_Ioctl::ReadSectors(PhysPt buffer,
 		if ((Bitu)ret != buflen)
 			ret = -1;
 	}
-	close(cdrom_fd);
 
 	MEM_BlockWrite(buffer, buf.data(), buflen);
 
 	return (ret > 0);
 }
 
-bool CDROM_Interface_Ioctl::SetDevice(const char *path, const int cd_number)
+bool CDROM_Interface_Ioctl::Open(const char* device_name)
 {
-	bool success = CDROM_Interface_SDL::SetDevice(path, cd_number);
+	int fd = open(device_name, O_RDONLY | O_NONBLOCK);
+	if (fd == -1) {
+		return false;
+	}
+	if (IsOpen()) {
+		close(cdrom_fd);
+	}
+	cdrom_fd = fd;
+	return true;
```

Comment:
Maybe put a comment there that `Open` also closes the currently open device (so effectively it "re-opens").